2020/day3.clj

```diff
@@ -38,52 +50,63 @@
 (def slopes [[1 1] [3 1] [5 1] [7 1] [1 2]])
 
 (comment
-  (do (def sample-input (file->seq "resources/2020/day3/input-sample.txt"))
-      sample-input)
+  (do (def sample-grid (file->seq "resources/2020/day3/input-sample.txt"))
+      sample-grid)
   #_=> ["..##......."
         ...
         ".#..#...#.#"]
 
+  (->> (locations-s [3 1])
+       (take 3))
+  #_=> ([0 0] [3 1] [6 2])
+
   (take-while neg? [-2 0 1])
   #_=> (-2)
 
-  (take 3 (map #(next-location [0 0] [(* 3 %) (* 1 %)])
-               (range)))
-  #_=> ([0 0] [3 1] [6 2])
+  (valid-locations (count sample-grid) (take 5 (locations-s [3 1])))
+  #_=> ([0 0] [3 1] [6 2] [9 3] [12 4])
 
-  (locations-s 3 [3 1])
-  #_=> ([0 0] [3 1] [6 2])
+  (nth ["a" "b" "c"] 1)
+  #_=> "b"
 
-  (mod 10 5)
-  #_=> 0
+  (char-at-location sample-grid [0 0])
+  #_=> \.
 
-  (get-in sample-input [0 2])
+  (char-at-location sample-grid [2 0])
   #_=> \#
 
-  (get-in sample-input [0 0])
-  #_=> \.
-
   (tree? \.)
   #_=> false
 
   (tree? \#)
   #_=> true
 
-  (count-trees-on-slope sample-input [3 1])
+  (def valid-locations-sample (valid-locations (count sample-grid) (locations-s [3 1])))
+  #_=> #'advent-of-code.2020.day3/valid-locations-sample
+
+  (println valid-locations-sample)
+  #_([0 0] [3 1] [6 2] [9 3] [12 4] [15 5] [18 6] [21 7] [24 8] [27 9] [30 10])
+
+  (->> valid-locations-sample
+       (map (fn [loc] (char-at-location sample-grid loc)))
```

Comment:
this is a partial

Reply:
> this is a partial

Do you mean we need to test the completed fn instead of the partial?

Reply:
https://clojuredocs.org/clojure.core/partial

Reply:
Written out is fine, but you don't need it here, since the preceeding term (`valid-locations-sample`) is already quite explicit